### PR TITLE
Bug fix in mercator coordinator transformation

### PR DESCRIPTION
**Pull request by GernotMaier**
Fixed call to set coordinates of a telescopes when input is in Mercator in [simtools/layout/array_layout.py](simtools/layout/array_layout.py).

Includes also 

- improvements in both units and integration tests in this context. 
- renaming of `alt` to `altitude` in some remaining places (these had been missed before)
- renaming of 'CORSIKA' to 'Ground' in some remaining places (had also been missed before)

Note that we hardly use the longitude/latitude system (essentially only to get the coordinates of the reference positions, and we will not use them for telescope coordinates. Still, for testing we are doing the conversion - to allow this to work I introduced a flag `skip_input_validation`, as we don't want to introduce a schema for mercator just for testing.

### CODE REVIEW

File: tests/unit_tests/layout/test_telescope_position.py [hunk 1]
```diff
@@ -55,15 +55,15 @@ def test_str(crs_wgs84, crs_local, crs_utm):
...
+    _test_string = "LST-01\t Ground x(->North): 50.00 y(->West): -25.00"
     assert _tcors == (_test_string + "\t Alt: 2158.00")
     tel.convert_all(crs_local=crs_local, crs_wgs84=crs_wgs84)
     _tcors = tel.__str__()
     _test_string += "\t Longitude: 28.76262 Latitude: -17.89177"
     assert _tcors == (_test_string + "\t Alt: 2158.00")
     tel.convert_all(crs_local=crs_local, crs_wgs84=crs_wgs84, crs_utm=crs_utm)
     _tcors = tel.__str__()
-    _test_string = "LST-01\t CORSIKA x(->North): 50.00 y(->West): -25.00"
+    _test_string = "LST-01\t Ground x(->North): 50.00 y(->West): -25.00"
```
Comment (VictorBarbosaMartins) [inline tests/unit_tests/layout/test_telescope_position.py:66]: I don't remember whether we discussed about replacing CORSIKA by Ground but I am afraid Ground does not pass the same information as CORSIKA. CORSIKA already comes with the definition of the coordinate system, where X points towards North and Ground is somewhat vague. If we manage to defined Ground somewhere to consider the coordinate system as CORSIKA I would agree with this chance.
Screenshot 2024-01-23 at 13 40 55

File: simtools/applications/print_array_elements.py [hunk 2]
```diff
@@ -145,6 +145,13 @@ def _parse(label=None, description=None):
         default=None,
         nargs="+",
     )
+    config.parser.add_argument(
```
Comment (VictorBarbosaMartins) [inline simtools/applications/print_array_elements.py:148]: Not sure it justifies adding a parameter to skip the validation, which is necessary only in the very specific case of longitude/latitude? Isn't it better to implement an if statement in the code and exclude the validation when longitude/latitude is used?

**Review by VictorBarbosaMartins — COMMENTED** [review]
Thanks for implementing this (and increasing coverage of both integration and unit tests). I leave two main comments proposing discussions.

File: tests/unit_tests/layout/test_telescope_position.py (hunk 1 above)
Comment (GernotMaier) [reply tests/unit_tests/layout/test_telescope_position.py:66]: The name CORSIKA is wrong, as this is not the CORSIKA coordinate system. Ours is aligned to geographic north, while CORSIKA is aligned with magnetic north.

This was discussed in https://github.com/gammasim/simtools/issues/658 and we have changed it everywhere else in the code.

File: simtools/applications/print_array_elements.py (hunk 2 above)
Comment (GernotMaier) [reply simtools/applications/print_array_elements.py:148]: I do think this is useful for other cases. We use this application a lot with input from quickly generated files. In many cases, the metadata is not complete (which is fine, as these are temporary fine) and maybe not all columns are there. So this is useful to have for this application.

File: tests/unit_tests/layout/test_telescope_position.py (hunk 1 above)
Comment (VictorBarbosaMartins) [reply tests/unit_tests/layout/test_telescope_position.py:66]: ok!

File: simtools/applications/print_array_elements.py (hunk 2 above)
Comment (VictorBarbosaMartins) [reply simtools/applications/print_array_elements.py:148]: ok. this should not become default though.

**Review by VictorBarbosaMartins — APPROVED** [review]

**Comment by GernotMaier** [comment]
@VictorBarbosaMartins - thanks for the quick review!

**Pull request merged**